Extension of QDltPluginControlInterface with new notification events

svlad-90 opened the pull request:
- Add possibility to inject message decoder facade into the
plugin
Motivation: allow plugin to decode messages, which it retrieves
from the QDltFile.
- Add possibility to inject main table view into the
plugin
Motivation: allow plugin to scroll main table view to a certain
row
- Add possibility to notify the plugin about the changed
configuration
Motivation: plugin should be able to reset its analysis
data in case of changed configuration
- Extend comments of QDltPluginViewerInterface::initMsg &
QDltPluginViewerInterface::initMsgDecoded methods in order to
reflect its cross-thread calling nature

All done changes were tested on a
self-made plugin.
Decoding of messages works as expected
Scrolling of
main table works as expected
Reaction on certain configuration-change
use-cases works as expected:
- filtering enabled & disabled
- filter
settings changed
- sort by time enabled & disabled

Signed-off-by: Vladyslav Goncharuk <svlad1990@gmail.com>

alexmucde commented:
Hi @svlad-90,

i tried your PR and it works, I have some comments:

- Your new plugin functions matches quiet well to the QDltPluginControlInterface. Perhaps we can move them there. A little problem is that all existing Plugins must be extended.
- I used the naming init instead of inject for the plugin functions, Perhaps you can rename them, so that they are more consistant

I will not be available the next few weeks, so do not wonder if things are not getting forward.

    Alex

alexmucde commented:
Additionaly we need a signed of statement in your commit message.

svlad-90 commented:
Hi @alexmucde, I will try to consider your findings during next weeks ( when you'll be absent ) and update the PR.
I'll notify you then.

- I will definitely change the names of methods.
- Regarding extension QDltPluginControlInterface - that is also acceptable.
- Also I'll check the other PR-s to check what is a "signed of statement"

Please, keep this PR opened until further update.
----

I will not be wondered by delayed response from your side, as I have all my internal issues solved even without this patch via the QT-based workarounds ☺

svlad-90 commented:
Hi @alexmucde, I've fixed all findings:
- Moved all new notifications into the QDltPluginControlInterface 
- Changed injectXXX to initXXX names

Together with that I've extended comments of QDltPluginViewerInterface::initMsg &
QDltPluginViewerInterface::initMsgDecoded methods in order to
reflect its cross-thread calling nature. Related to the following issue:
https://github.com/GENIVI/dlt-viewer/issues/50

=> Please, review once again

svlad-90 commented:
... ping ... anybody here?

svlad-90 commented:
... ping ... anybody here?

gunnarx commented:
@svlad-90 One thing - Signoff line is always at the end.
If you're interested in more recommendations, [this page](https://at.projects.genivi.org/wiki/display/PROJ/Rules+for+Commit+Messages) might help. 
(This is typical vacation time in Germany - this might be why the maintainer @alexmucde can't reply right away).

alexmucde commented:
Sorry that i cannot work immediatly on this topic, but i will not forget you. I have other high priority topics.

Next steps will be:
- Anaylse your changes, then i will give feedback
- Testbuild
- Integration

And as Gunnar said please add always a signed off statement to the commit message.

svlad-90 commented:
Hi @gunnarx, as you've requested, I've moved the signed off statement to the bottom of both commit message & PR description. 

Also no any offences regarding delayed reply. I understand, that maintainers also spend their own time to review the proposed changes.

----

Hi @alexmucde, I've fixed the signed off statement, and understood next steps. Looking forward to get further feedback from you ☺

svlad-90 commented:
2 weeks later ... ping ... anybody here? ☺

alexmucde commented:
Sorry Gernot and me still very busy with other topics. I cannot promise you, when we find some time for DLT Viewer.

alexmucde commented:
Hi @svlad-90,

Sorry for the delay.
Changes looks good to me now. We will now test this pull request before merging it.

   Alex

alexmucde commented:
Tested build internally, will now merge.
Again sorry for the delay.
